src/shared/sandboxRequest.ts

```diff
@@ -33,13 +33,28 @@ export async function createSandboxRequest(
   definitionFile: string | undefined,
   logger?: Logger | undefined,
   properties?: Record<string, string | undefined>
-): Promise<{ sandboxReq: SandboxRequest; srcSandboxName: string }>;
+): Promise<{
+  sandboxReq: SandboxRequest & {
+    ApexClassName: string | undefined;
+    ApexClassId: string | undefined;
+    ActivationUserGroupName: string | undefined;
+    ActivationUserGroupId: string | undefined;
```

Comment:
Of these 4 new fields, `ApexClassId` and `ActivationUserGroupId` are real properties in the sandbox request, let's add them here instead:
https://github.com/forcedotcom/sfdx-core/blob/4580a1c323379292e6da1ddb0e00458782ca1b51/src/org/org.ts#L128

`SandboxRequest` represents the payload for a sandbox create/refresh request so we don't want to add the apexclass/activationusergroup `Name` fields there (they only exists in the definition file that we handle)